File: internal/content/testdata/bench/javascript/nc-valid-paren-string.js
function check_valid_string(s) {
    let lo = 0, hi = 0;
    for (const c of s) {
        if (c === '(') { lo++; hi++; }
        else if (c === ')') { lo--; hi--; }
        else { lo--; hi++; }
        if (hi < 0) return false;
        if (lo < 0) lo = 0;
    }
    return lo === 0;
}

File: internal/content/testdata/bench/go/nc-valid-paren-string.go
func check_valid_string(s string) bool {
	lo, hi := 0, 0
	for _, c := range s {
		if c == '(' {
			lo++
			hi++
		} else if c == ')' {
			lo--
			hi--
		} else {
			lo--
			hi++
		}
		if hi < 0 {
			return false
		}
		if lo < 0 {
			lo = 0
		}
	}
	return lo == 0
}
